src/ActionList/List.tsx

```diff
@@ -0,0 +1,146 @@
+import {Group, GroupProps} from './Group'
+import {Header, HeaderProps} from './Header'
+import {Item, ItemProps} from './Item'
+
+import {Divider} from './Divider'
+import type {Flatten} from '../utils/types'
+import React from 'react'
+import {get} from '../constants'
+import styled from 'styled-components'
+
+/**
+ * Contract for props passed to the `List` component.
+ */
+export interface UngroupedListProps {
+  /**
+   * `Item`s to render in the `List`.
+   */
+  items: ItemProps[]
+
+  /**
+   * A `List`-level custom `Item` renderer. Every `Item` within this `List`
+   * without a `Group`-level or `Item`-level custom `Item` renderer will be
+   * rendered using this function component.
+   */
+  renderItem?: (props: ItemProps) => JSX.Element
+
+  /**
+   * An option function to run after running an item's onClick function -
+   * because items can have their own specific click handler, this function
+   * allows global clean up tasks such as a menu hide after a selection is made
+   */
+  afterSelect?: (e: Event) => void
```

Comment:
I think there is a necessity for this, but I'm curious to know thoughts if there are better ways to accomplish this?

Basically, since we can define click handlers on individual items, we need a more global method that can be called after these more specific methods to accomplish tasks like hiding a menu overlay.